programming/sed-article.md

```diff
@@ -2,88 +2,75 @@
 
 ## Overview
 
-`sed` is a Unix utility that transforms text.
-The name is short for "stream editor".
-Input comes from a stream of text and
-output is an edited version of that text.
+`sed` is a Unix text-processing utility.
+The name `sed` is short for "stream editor".
+Like other filters in the Unix pipe-and-filter paradigm, the input to `sed` is a stream of text and the output is an edited version of that text.
+`sed` has been branded a "power tool" and "Swiss Army Knife" and while most developers don't need a stream editor
+on a daily basis, `sed` is a great tool for your automated text manipulation toolbox.
 
-While most developers don't need a stream editor
-on a daily basis, `sed` is a great tool belt addition.
-
-Memorizing all the capabilities and syntax details
-of `sed` can be a daunting task.
-But after reading this article you will know enough
-to recognize when using `sed` is appropriate
-and can come back to it later for a refresher.
+After reading this article you will know enough
+to recognize when using `sed` is appropriate and will be familiar with the features of `sed` that are used most often in practice.
+When necessary, you can return to this article for a refresher on the more advanced features of `sed`.
 
 `sed` is based on the `ed` editor
 which was created by Ken Thompson in 1969.
 `sed` was created Lee McMahon of Bell Labs in 1973.
-Currently the most popular version is GNU sed.
+Currently, the most popular version is GNU sed.
 This is available on all major operating systems
 including Linux, macOS, and Windows.
 
-Alternatives to `sed` include the Unix utility `awk`
+Alternatives to `sed` include the Unix utility `awk`, `perl`,
 and nearly any general purpose programming language.
 
-Awk excels at processing record-based text
+`awk` excels at processing record-based text
 where each line is thought of as a record
 composed of columns with similar formatting,
 perhaps separated by spaces or tabs.
 
 General purpose programming languages have the advantage
 that they can perform any kind of text transformation.
-But the code required is longer than when using `sed`.
```

Comment:
Why do you think I shouldn't point out that if you choose to do text transformation in a general purpose language, you will write more code?

Reply:
When I'm using sed, I'm less concerned with the length of the program than I am with the time it takes to develop the program.  What about

However, much of the program in a general purpose language will reimplement many of the things that `sed` is already doing.

Reply:
I like it!